{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "scalaabsa_thesis_working.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "mount_file_id": "1a3U01UVcNHeyodFHHa8AqNnFP0CQghqp",
      "authorship_tag": "ABX9TyN2QExYaIquEeY2kbR1PSRt",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "accelerator": "GPU"
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/azzollinigianluca95/thesis_1/blob/main/scalaabsa_thesis_working_4.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "s_4HeDuBGcnY",
        "outputId": "e710e1cc-ffd3-46af-fd66-b9f13f318627"
      },
      "source": [
        "!git clone https://github.com/ScalaConsultants/Aspect-Based-Sentiment-Analysis.git\r\n"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Cloning into 'Aspect-Based-Sentiment-Analysis'...\n",
            "remote: Enumerating objects: 357, done.\u001b[K\n",
            "remote: Counting objects: 100% (357/357), done.\u001b[K\n",
            "remote: Compressing objects: 100% (154/154), done.\u001b[K\n",
            "remote: Total 1238 (delta 216), reused 310 (delta 198), pack-reused 881\u001b[K\n",
            "Receiving objects: 100% (1238/1238), 1.89 MiB | 28.95 MiB/s, done.\n",
            "Resolving deltas: 100% (755/755), done.\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8-s3gEMpGqAf",
        "outputId": "778f564c-d21f-405d-a884-6af96218f1a8"
      },
      "source": [
        "#A)\r\n",
        "#ARTICLE: https://towardsdatascience.com/conda-google-colab-75f7c867a522\r\n",
        "!python --version\r\n",
        "#B)\r\n",
        "!wget -c https://repo.anaconda.com/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh\r\n",
        "!chmod +x Miniconda3-4.5.4-Linux-x86_64.sh\r\n",
        "!bash ./Miniconda3-4.5.4-Linux-x86_64.sh -b -f -p /usr/local\r\n",
        "# update 1\r\n",
        "!conda install -q -y --prefix /usr/local python=3.6 ujson\r\n",
        "# update 2\r\n",
        "import sys\r\n",
        "sys.path.append('/usr/local/lib/python3.6/site-packages')\r\n",
        "# test it\r\n",
        "import ujson\r\n",
        "print(ujson.dumps({1:2}))\r\n",
        "#C)\r\n",
        "!which conda\r\n",
        "#D)\r\n",
        "!conda --version\r\n",
        "!which python\r\n",
        "!python --version\r\n",
        "!conda install --channel defaults conda python=3.6 --yes\r\n",
        "!conda update --channel defaults --all --yes\r\n",
        "#E)\r\n",
        "!conda --version # now returns 4.8.3\r\n",
        "!python --version # now returns Python 3.6.10 :: Anaconda, Inc.\r\n",
        "#F)\r\n",
        "import sys\r\n",
        "sys.path\r\n",
        "#F)\r\n",
        "'''Now all you need to do is install you favorite packages. Just remember \r\n",
        "to include the --yes flag when installing your packages to avoid getting \r\n",
        "prompted to confirm the package plan.'''\r\n",
        "!conda install --channel conda-forge featuretools --yes"
      ],
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Python 3.6.12 :: Anaconda, Inc.\n",
            "--2020-12-27 18:41:13--  https://repo.anaconda.com/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh\n",
            "Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.131.3, 104.16.130.3, 2606:4700::6810:8203, ...\n",
            "Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.131.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 416 Requested Range Not Satisfiable\n",
            "\n",
            "    The file is already fully retrieved; nothing to do.\n",
            "\n",
            "PREFIX=/usr/local\n",
            "installing: python-3.6.5-hc3d631a_2 ...\n",
            "Python 3.6.5 :: Anaconda, Inc.\n",
            "installing: ca-certificates-2018.03.07-0 ...\n",
            "installing: conda-env-2.6.0-h36134e3_1 ...\n",
            "installing: libgcc-ng-7.2.0-hdf63c60_3 ...\n",
            "installing: libstdcxx-ng-7.2.0-hdf63c60_3 ...\n",
            "installing: libffi-3.2.1-hd88cf55_4 ...\n",
            "installing: ncurses-6.1-hf484d3e_0 ...\n",
            "installing: openssl-1.0.2o-h20670df_0 ...\n",
            "installing: tk-8.6.7-hc745277_3 ...\n",
            "installing: xz-5.2.4-h14c3975_4 ...\n",
            "installing: yaml-0.1.7-had09818_2 ...\n",
            "installing: zlib-1.2.11-ha838bed_2 ...\n",
            "installing: libedit-3.1.20170329-h6b74fdf_2 ...\n",
            "installing: readline-7.0-ha6073c6_4 ...\n",
            "installing: sqlite-3.23.1-he433501_0 ...\n",
            "installing: asn1crypto-0.24.0-py36_0 ...\n",
            "installing: certifi-2018.4.16-py36_0 ...\n",
            "installing: chardet-3.0.4-py36h0f667ec_1 ...\n",
            "installing: idna-2.6-py36h82fb2a8_1 ...\n",
            "installing: pycosat-0.6.3-py36h0a5515d_0 ...\n",
            "installing: pycparser-2.18-py36hf9f622e_1 ...\n",
            "installing: pysocks-1.6.8-py36_0 ...\n",
            "installing: ruamel_yaml-0.15.37-py36h14c3975_2 ...\n",
            "installing: six-1.11.0-py36h372c433_1 ...\n",
            "installing: cffi-1.11.5-py36h9745a5d_0 ...\n",
            "installing: setuptools-39.2.0-py36_0 ...\n",
            "installing: cryptography-2.2.2-py36h14c3975_0 ...\n",
            "installing: wheel-0.31.1-py36_0 ...\n",
            "installing: pip-10.0.1-py36_0 ...\n",
            "installing: pyopenssl-18.0.0-py36_0 ...\n",
            "installing: urllib3-1.22-py36hbe7ace6_0 ...\n",
            "installing: requests-2.18.4-py36he2e5f8d_1 ...\n",
            "installing: conda-4.5.4-py36_0 ...\n",
            "unlinking: ca-certificates-2020.12.5-ha878542_0\n",
            "unlinking: certifi-2020.12.5-py36h5fab9bb_0\n",
            "unlinking: cffi-1.14.4-py36h261ae71_0\n",
            "unlinking: chardet-4.0.0-py36h06a4308_1003\n",
            "unlinking: conda-4.9.2-py36h5fab9bb_0\n",
            "unlinking: cryptography-3.3.1-py36h3c74f83_0\n",
            "unlinking: idna-2.10-py_0\n",
            "unlinking: libedit-3.1.20191231-h14c3975_1\n",
            "unlinking: libffi-3.3-he6710b0_2\n",
            "unlinking: libgcc-ng-9.1.0-hdf63c60_0\n",
            "unlinking: libstdcxx-ng-9.1.0-hdf63c60_0\n",
            "unlinking: ncurses-6.2-he6710b0_1\n",
            "unlinking: openssl-1.1.1i-h27cfd23_0\n",
            "unlinking: pip-20.3.3-py36h06a4308_0\n",
            "unlinking: pycosat-0.6.3-py36h27cfd23_0\n",
            "unlinking: pycparser-2.20-py_2\n",
            "unlinking: pyopenssl-20.0.1-pyhd3eb1b0_1\n",
            "unlinking: pysocks-1.7.1-py36h06a4308_0\n",
            "unlinking: python-3.6.12-hcff3b4d_2\n",
            "unlinking: readline-8.0-h7b6447c_0\n",
            "unlinking: requests-2.25.1-pyhd3eb1b0_0\n",
            "unlinking: ruamel_yaml-0.15.87-py36h7b6447c_1\n",
            "unlinking: setuptools-51.0.0-py36h06a4308_2\n",
            "unlinking: six-1.15.0-py36h06a4308_0\n",
            "unlinking: sqlite-3.33.0-h62c20be_0\n",
            "unlinking: tk-8.6.10-hbc83047_0\n",
            "unlinking: urllib3-1.26.2-pyhd3eb1b0_0\n",
            "unlinking: wheel-0.36.2-pyhd3eb1b0_0\n",
            "unlinking: xz-5.2.5-h7b6447c_0\n",
            "unlinking: yaml-0.2.5-h7b6447c_0\n",
            "unlinking: zlib-1.2.11-h7b6447c_3\n",
            "installation finished.\n",
            "WARNING:\n",
            "    You currently have a PYTHONPATH environment variable set. This may cause\n",
            "    unexpected behavior when running the Python interpreter in Miniconda3.\n",
            "    For best results, please verify that your PYTHONPATH only points to\n",
            "    directories of packages that are compatible with the Python interpreter\n",
            "    in Miniconda3: /usr/local\n",
            "Solving environment: ...working... done\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs: \n",
            "    - python=3.6\n",
            "    - ujson\n",
            "\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "    ca-certificates: 2018.03.07-0            --> 2020.12.8-h06a4308_0    \n",
            "    certifi:         2018.4.16-py36_0        --> 2020.12.5-py36h06a4308_0\n",
            "    cffi:            1.11.5-py36h9745a5d_0   --> 1.14.4-py36h261ae71_0   \n",
            "    conda:           4.5.4-py36_0            --> 4.9.2-py36h06a4308_0    \n",
            "    cryptography:    2.2.2-py36h14c3975_0    --> 3.3.1-py36h3c74f83_0    \n",
            "    libedit:         3.1.20170329-h6b74fdf_2 --> 3.1.20191231-h14c3975_1 \n",
            "    libffi:          3.2.1-hd88cf55_4        --> 3.3-he6710b0_2          \n",
            "    libgcc-ng:       7.2.0-hdf63c60_3        --> 9.1.0-hdf63c60_0        \n",
            "    libstdcxx-ng:    7.2.0-hdf63c60_3        --> 9.1.0-hdf63c60_0        \n",
            "    ncurses:         6.1-hf484d3e_0          --> 6.2-he6710b0_1          \n",
            "    openssl:         1.0.2o-h20670df_0       --> 1.1.1i-h27cfd23_0       \n",
            "    python:          3.6.5-hc3d631a_2        --> 3.6.12-hcff3b4d_2       \n",
            "    readline:        7.0-ha6073c6_4          --> 8.0-h7b6447c_0          \n",
            "    sqlite:          3.23.1-he433501_0       --> 3.33.0-h62c20be_0       \n",
            "    tk:              8.6.7-hc745277_3        --> 8.6.10-hbc83047_0       \n",
            "    xz:              5.2.4-h14c3975_4        --> 5.2.5-h7b6447c_0        \n",
            "\n",
            "Preparing transaction: ...working... done\n",
            "Verifying transaction: ...working... done\n",
            "Executing transaction: ...working... done\n",
            "{\"1\":2}\n",
            "/usr/local/bin/conda\n",
            "conda 4.9.2\n",
            "/usr/local/bin/python\n",
            "Python 3.6.12 :: Anaconda, Inc.\n",
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \n",
            "The environment is inconsistent, please check the package plan carefully\n",
            "The following packages are causing the inconsistency:\n",
            "\n",
            "  - conda-forge/linux-64::pyyaml==5.3.1=py36he6145b8_1\n",
            "  - conda-forge/noarch::dask-core==2020.12.0=pyhd8ed1ab_0\n",
            "  - conda-forge/noarch::featuretools==0.22.0=pyhd8ed1ab_0\n",
            "  - conda-forge/noarch::dask==2020.12.0=pyhd8ed1ab_0\n",
            "  - conda-forge/linux-64::bokeh==1.4.0=py36h9f0ad1d_1\n",
            "  - conda-forge/linux-64::distributed==2020.12.0=py36h5fab9bb_0\n",
            "\b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - conda\n",
            "    - python=3.6\n",
            "\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ruamel_yaml                        0.15.37-py36h14c3975_2 --> 0.15.87-py36h7b6447c_1\n",
            "  yaml                                     0.1.7-had09818_2 --> 0.2.5-h7b6447c_0\n",
            "\n",
            "\n",
            "Preparing transaction: - \b\bdone\n",
            "Verifying transaction: | \b\bdone\n",
            "Executing transaction: - \b\bdone\n",
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "\n",
            "The following packages will be REMOVED:\n",
            "\n",
            "  asn1crypto-0.24.0-py36_0\n",
            "  bokeh-1.4.0-py36h9f0ad1d_1\n",
            "  click-7.1.2-pyh9f0ad1d_0\n",
            "  cloudpickle-1.6.0-py_0\n",
            "  conda-env-2.6.0-h36134e3_1\n",
            "  contextvars-2.4-py_0\n",
            "  cytoolz-0.11.0-py36h1d69622_1\n",
            "  dask-2020.12.0-pyhd8ed1ab_0\n",
            "  dask-core-2020.12.0-pyhd8ed1ab_0\n",
            "  distributed-2020.12.0-py36h5fab9bb_0\n",
            "  featuretools-0.22.0-pyhd8ed1ab_0\n",
            "  freetype-2.10.4-h7ca028e_0\n",
            "  fsspec-0.8.5-pyhd8ed1ab_0\n",
            "  heapdict-1.0.1-py_0\n",
            "  immutables-0.14-py36h8c4c3a4_1\n",
            "  jinja2-2.11.2-pyh9f0ad1d_0\n",
            "  jpeg-9d-h36c2ea0_0\n",
            "  libblas-3.9.0-3_openblas\n",
            "  libcblas-3.9.0-3_openblas\n",
            "  libgfortran-ng-7.5.0-hae1eefd_17\n",
            "  libgfortran4-7.5.0-hae1eefd_17\n",
            "  liblapack-3.9.0-3_openblas\n",
            "  libopenblas-0.3.12-pthreads_hb3c22a3_1\n",
            "  libpng-1.6.37-h21135ba_2\n",
            "  libtiff-4.0.10-hc3755c2_1005\n",
            "  locket-0.2.0-py_2\n",
            "  lz4-c-1.9.2-he1b5a44_3\n",
            "  markupsafe-1.1.1-py36he6145b8_2\n",
            "  msgpack-python-1.0.1-py36hff7bd54_0\n",
            "  numpy-1.19.4-py36h8732dcd_1\n",
            "  olefile-0.46-pyh9f0ad1d_1\n",
            "  packaging-20.8-pyhd3deb0d_0\n",
            "  pandas-1.1.4-py36hd87012b_0\n",
            "  partd-1.1.0-py_0\n",
            "  pillow-6.2.1-py36h6b7be26_0\n",
            "  psutil-5.4.8-py36h14c3975_1000\n",
            "  pyparsing-2.4.7-pyh9f0ad1d_0\n",
            "  python-dateutil-2.8.1-py_0\n",
            "  python_abi-3.6-1_cp36m\n",
            "  pytz-2020.5-pyhd8ed1ab_0\n",
            "  pyyaml-5.3.1-py36he6145b8_1\n",
            "  scipy-1.5.3-py36h976291a_0\n",
            "  sortedcontainers-2.3.0-pyhd8ed1ab_0\n",
            "  tblib-1.6.0-py_0\n",
            "  toolz-0.11.1-py_0\n",
            "  tornado-6.1-py36h1d69622_0\n",
            "  zict-2.0.0-py_0\n",
            "  zstd-1.4.5-h6597ccf_2\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  chardet                              3.0.4-py36h0f667ec_1 --> 4.0.0-py36h06a4308_1003\n",
            "  idna               pkgs/main/linux-64::idna-2.6-py36h82f~ --> pkgs/main/noarch::idna-2.10-py_0\n",
            "  pip                                         10.0.1-py36_0 --> 20.3.3-py36h06a4308_0\n",
            "  pycparser          pkgs/main/linux-64::pycparser-2.18-py~ --> pkgs/main/noarch::pycparser-2.20-py_2\n",
            "  pyopenssl          pkgs/main/linux-64::pyopenssl-18.0.0-~ --> pkgs/main/noarch::pyopenssl-20.0.1-pyhd3eb1b0_1\n",
            "  pysocks                                      1.6.8-py36_0 --> 1.7.1-py36h06a4308_0\n",
            "  requests           pkgs/main/linux-64::requests-2.18.4-p~ --> pkgs/main/noarch::requests-2.25.1-pyhd3eb1b0_0\n",
            "  setuptools                                  39.2.0-py36_0 --> 51.0.0-py36h06a4308_2\n",
            "  six                                 1.11.0-py36h372c433_1 --> 1.15.0-py36h06a4308_0\n",
            "  urllib3            pkgs/main/linux-64::urllib3-1.22-py36~ --> pkgs/main/noarch::urllib3-1.26.2-pyhd3eb1b0_0\n",
            "  wheel              pkgs/main/linux-64::wheel-0.31.1-py36~ --> pkgs/main/noarch::wheel-0.36.2-pyhd3eb1b0_0\n",
            "  zlib                                    1.2.11-ha838bed_2 --> 1.2.11-h7b6447c_3\n",
            "\n",
            "The following packages will be DOWNGRADED:\n",
            "\n",
            "  pycosat                              0.6.3-py36h0a5515d_0 --> 0.6.3-py36h27cfd23_0\n",
            "\n",
            "\n",
            "Preparing transaction: / \b\b- \b\bdone\n",
            "Verifying transaction: | \b\b/ \b\bdone\n",
            "Executing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "conda 4.9.2\n",
            "Python 3.6.12 :: Anaconda, Inc.\n",
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - featuretools\n",
            "\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bokeh              conda-forge/linux-64::bokeh-1.4.0-py36h9f0ad1d_1\n",
            "  click              conda-forge/noarch::click-7.1.2-pyh9f0ad1d_0\n",
            "  cloudpickle        conda-forge/noarch::cloudpickle-1.6.0-py_0\n",
            "  contextvars        conda-forge/noarch::contextvars-2.4-py_0\n",
            "  cytoolz            conda-forge/linux-64::cytoolz-0.11.0-py36h1d69622_1\n",
            "  dask               conda-forge/noarch::dask-2020.12.0-pyhd8ed1ab_0\n",
            "  dask-core          conda-forge/noarch::dask-core-2020.12.0-pyhd8ed1ab_0\n",
            "  distributed        conda-forge/linux-64::distributed-2020.12.0-py36h5fab9bb_0\n",
            "  featuretools       conda-forge/noarch::featuretools-0.22.0-pyhd8ed1ab_0\n",
            "  freetype           conda-forge/linux-64::freetype-2.10.4-h7ca028e_0\n",
            "  fsspec             conda-forge/noarch::fsspec-0.8.5-pyhd8ed1ab_0\n",
            "  heapdict           conda-forge/noarch::heapdict-1.0.1-py_0\n",
            "  immutables         conda-forge/linux-64::immutables-0.14-py36h8c4c3a4_1\n",
            "  jinja2             conda-forge/noarch::jinja2-2.11.2-pyh9f0ad1d_0\n",
            "  jpeg               conda-forge/linux-64::jpeg-9d-h36c2ea0_0\n",
            "  libblas            conda-forge/linux-64::libblas-3.9.0-3_openblas\n",
            "  libcblas           conda-forge/linux-64::libcblas-3.9.0-3_openblas\n",
            "  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.5.0-hae1eefd_17\n",
            "  libgfortran4       conda-forge/linux-64::libgfortran4-7.5.0-hae1eefd_17\n",
            "  liblapack          conda-forge/linux-64::liblapack-3.9.0-3_openblas\n",
            "  libopenblas        conda-forge/linux-64::libopenblas-0.3.12-pthreads_hb3c22a3_1\n",
            "  libpng             conda-forge/linux-64::libpng-1.6.37-h21135ba_2\n",
            "  libtiff            conda-forge/linux-64::libtiff-4.0.10-hc3755c2_1005\n",
            "  locket             conda-forge/noarch::locket-0.2.0-py_2\n",
            "  lz4-c              conda-forge/linux-64::lz4-c-1.9.2-he1b5a44_3\n",
            "  markupsafe         conda-forge/linux-64::markupsafe-1.1.1-py36he6145b8_2\n",
            "  msgpack-python     pkgs/main/linux-64::msgpack-python-1.0.1-py36hff7bd54_0\n",
            "  numpy              conda-forge/linux-64::numpy-1.19.4-py36h8732dcd_1\n",
            "  olefile            conda-forge/noarch::olefile-0.46-pyh9f0ad1d_1\n",
            "  packaging          conda-forge/noarch::packaging-20.8-pyhd3deb0d_0\n",
            "  pandas             conda-forge/linux-64::pandas-1.1.4-py36hd87012b_0\n",
            "  partd              conda-forge/noarch::partd-1.1.0-py_0\n",
            "  pillow             conda-forge/linux-64::pillow-6.2.1-py36h6b7be26_0\n",
            "  psutil             conda-forge/linux-64::psutil-5.4.8-py36h14c3975_1000\n",
            "  pyparsing          conda-forge/noarch::pyparsing-2.4.7-pyh9f0ad1d_0\n",
            "  python-dateutil    conda-forge/noarch::python-dateutil-2.8.1-py_0\n",
            "  python_abi         conda-forge/linux-64::python_abi-3.6-1_cp36m\n",
            "  pytz               conda-forge/noarch::pytz-2020.5-pyhd8ed1ab_0\n",
            "  pyyaml             conda-forge/linux-64::pyyaml-5.3.1-py36he6145b8_1\n",
            "  scipy              conda-forge/linux-64::scipy-1.5.3-py36h976291a_0\n",
            "  sortedcontainers   conda-forge/noarch::sortedcontainers-2.3.0-pyhd8ed1ab_0\n",
            "  tblib              conda-forge/noarch::tblib-1.6.0-py_0\n",
            "  toolz              conda-forge/noarch::toolz-0.11.1-py_0\n",
            "  tornado            conda-forge/linux-64::tornado-6.1-py36h1d69622_0\n",
            "  zict               conda-forge/noarch::zict-2.0.0-py_0\n",
            "  zstd               conda-forge/linux-64::zstd-1.4.5-h6597ccf_2\n",
            "\n",
            "The following packages will be SUPERSEDED by a higher-priority channel:\n",
            "\n",
            "  ca-certificates    pkgs/main::ca-certificates-2020.12.8-~ --> conda-forge::ca-certificates-2020.12.5-ha878542_0\n",
            "  certifi            pkgs/main::certifi-2020.12.5-py36h06a~ --> conda-forge::certifi-2020.12.5-py36h5fab9bb_0\n",
            "  conda               pkgs/main::conda-4.9.2-py36h06a4308_0 --> conda-forge::conda-4.9.2-py36h5fab9bb_0\n",
            "\n",
            "\n",
            "Preparing transaction: - \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Verifying transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Executing transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pKRieJCoitIz",
        "outputId": "e84638e8-2c86-476c-9035-3c7f130e3876"
      },
      "source": [
        "#CREATE ZIP OF A FOLDER:\r\n",
        "#!zip -r /content/file.zip /content/Aspect-Based-Sentiment-Analysis"
      ],
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "cp: cannot create regular file '/content/drive/file.zip': Operation not supported\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "08fLSIJYGrep",
        "outputId": "fde3a1c5-d85d-4fc1-ae14-6bc786379986"
      },
      "source": [
        "!conda env create -f=/content/Aspect-Based-Sentiment-Analysis/environment.yml #--force"
      ],
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Warning: you have pip-installed dependencies in your environment file, but you do not list pip itself as one of your conda dependencies.  Conda may not use the correct pip to install your packages, and they may end up in the wrong place.  Please add an explicit pip dependency.  I'm adding one for you, but still nagging you.\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \n",
            "Warning: 2 possible package resolutions (only showing differing packages):\n",
            "  - defaults/linux-64::jedi-0.17.0-py36_0, defaults/noarch::parso-0.8.1-pyhd3eb1b0_0\n",
            "  - defaults/linux-64::jedi-0.17.2-py36h06a4308_1, defaults/noarch::parso-0.7.0-py_0\b\bdone\n",
            "Preparing transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Verifying transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Installing pip dependencies: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| Ran pip subprocess with arguments:\n",
            "['/usr/local/envs/Aspect-Based-Sentiment-Analysis/bin/python', '-m', 'pip', 'install', '-U', '-r', '/content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt']\n",
            "Pip subprocess output:\n",
            "Collecting google-cloud-storage==1.26\n",
            "  Using cached google_cloud_storage-1.26.0-py2.py3-none-any.whl (75 kB)\n",
            "Collecting optuna==1.3\n",
            "  Using cached optuna-1.3.0-py3-none-any.whl\n",
            "Requirement already satisfied: numpy in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from optuna==1.3->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 5)) (1.19.2)\n",
            "Requirement already satisfied: scipy!=1.4.0 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from optuna==1.3->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 5)) (1.5.2)\n",
            "Requirement already satisfied: tqdm in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from optuna==1.3->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 5)) (4.54.1)\n",
            "Requirement already satisfied: joblib in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from optuna==1.3->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 5)) (1.0.0)\n",
            "Collecting tensorflow==2.2\n",
            "  Using cached tensorflow-2.2.0-cp36-cp36m-manylinux2010_x86_64.whl (516.2 MB)\n",
            "Requirement already satisfied: wheel>=0.26 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from tensorflow==2.2->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 1)) (0.36.2)\n",
            "Requirement already satisfied: six>=1.12.0 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from tensorflow==2.2->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 1)) (1.15.0)\n",
            "Collecting testfixtures==6.14\n",
            "  Using cached testfixtures-6.14.0-py2.py3-none-any.whl (88 kB)\n",
            "Collecting transformers==2.5\n",
            "  Using cached transformers-2.5.0-py3-none-any.whl (481 kB)\n",
            "Requirement already satisfied: requests in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from transformers==2.5->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 4)) (2.25.1)\n",
            "Collecting astunparse==1.6.3\n",
            "  Using cached astunparse-1.6.3-py2.py3-none-any.whl (12 kB)\n",
            "Collecting gast==0.3.3\n",
            "  Using cached gast-0.3.3-py2.py3-none-any.whl (9.7 kB)\n",
            "Collecting scipy!=1.4.0\n",
            "  Using cached scipy-1.4.1-cp36-cp36m-manylinux1_x86_64.whl (26.1 MB)\n",
            "Collecting tokenizers==0.5.0\n",
            "  Using cached tokenizers-0.5.0-cp36-cp36m-manylinux1_x86_64.whl (3.8 MB)\n",
            "Collecting absl-py>=0.7.0\n",
            "  Using cached absl_py-0.11.0-py3-none-any.whl (127 kB)\n",
            "Collecting google-auth<2.0dev,>=1.11.0\n",
            "  Using cached google_auth-1.24.0-py2.py3-none-any.whl (114 kB)\n",
            "Requirement already satisfied: setuptools>=40.3.0 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from google-auth<2.0dev,>=1.11.0->google-cloud-storage==1.26->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 2)) (51.0.0.post20201207)\n",
            "Collecting cachetools<5.0,>=2.0.0\n",
            "  Using cached cachetools-4.2.0-py3-none-any.whl (12 kB)\n",
            "Collecting google-cloud-core<2.0dev,>=1.2.0\n",
            "  Using cached google_cloud_core-1.5.0-py2.py3-none-any.whl (27 kB)\n",
            "Collecting google-api-core<2.0.0dev,>=1.21.0\n",
            "  Using cached google_api_core-1.24.1-py2.py3-none-any.whl (92 kB)\n",
            "Collecting google-pasta>=0.1.8\n",
            "  Using cached google_pasta-0.2.0-py3-none-any.whl (57 kB)\n",
            "Collecting google-resumable-media<0.6dev,>=0.5.0\n",
            "  Using cached google_resumable_media-0.5.1-py2.py3-none-any.whl (38 kB)\n",
            "Collecting googleapis-common-protos<2.0dev,>=1.6.0\n",
            "  Using cached googleapis_common_protos-1.52.0-py2.py3-none-any.whl (100 kB)\n",
            "Collecting grpcio>=1.8.6\n",
            "  Using cached grpcio-1.34.0-cp36-cp36m-manylinux2014_x86_64.whl (4.0 MB)\n",
            "Collecting h5py<2.11.0,>=2.10.0\n",
            "  Using cached h5py-2.10.0-cp36-cp36m-manylinux1_x86_64.whl (2.9 MB)\n",
            "Collecting keras-preprocessing>=1.1.0\n",
            "  Using cached Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)\n",
            "Collecting opt-einsum>=2.3.2\n",
            "  Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)\n",
            "Collecting protobuf>=3.8.0\n",
            "  Using cached protobuf-3.14.0-cp36-cp36m-manylinux1_x86_64.whl (1.0 MB)\n",
            "Collecting pyasn1-modules>=0.2.1\n",
            "  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)\n",
            "Collecting pyasn1<0.5.0,>=0.4.6\n",
            "  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)\n",
            "Collecting regex!=2019.12.17\n",
            "  Using cached regex-2020.11.13-cp36-cp36m-manylinux2014_x86_64.whl (723 kB)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from requests->transformers==2.5->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 4)) (1.26.2)\n",
            "Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from requests->transformers==2.5->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 4)) (4.0.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from requests->transformers==2.5->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 4)) (2020.12.5)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from requests->transformers==2.5->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 4)) (2.10)\n",
            "Collecting rsa<5,>=3.1.4\n",
            "  Using cached rsa-4.6-py3-none-any.whl (47 kB)\n",
            "Collecting sqlalchemy>=1.1.0\n",
            "  Using cached SQLAlchemy-1.3.22-cp36-cp36m-manylinux2010_x86_64.whl (1.3 MB)\n",
            "Collecting tensorboard<2.3.0,>=2.2.0\n",
            "  Using cached tensorboard-2.2.2-py3-none-any.whl (3.0 MB)\n",
            "Collecting google-auth-oauthlib<0.5,>=0.4.1\n",
            "  Using cached google_auth_oauthlib-0.4.2-py2.py3-none-any.whl (18 kB)\n",
            "Collecting markdown>=2.6.8\n",
            "  Using cached Markdown-3.3.3-py3-none-any.whl (96 kB)\n",
            "Requirement already satisfied: importlib-metadata in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from markdown>=2.6.8->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 1)) (2.0.0)\n",
            "Collecting requests-oauthlib>=0.7.0\n",
            "  Using cached requests_oauthlib-1.3.0-py2.py3-none-any.whl (23 kB)\n",
            "Collecting oauthlib>=3.0.0\n",
            "  Using cached oauthlib-3.1.0-py2.py3-none-any.whl (147 kB)\n",
            "Collecting tensorboard-plugin-wit>=1.6.0\n",
            "  Using cached tensorboard_plugin_wit-1.7.0-py3-none-any.whl (779 kB)\n",
            "Collecting tensorflow-estimator<2.3.0,>=2.2.0\n",
            "  Using cached tensorflow_estimator-2.2.0-py2.py3-none-any.whl (454 kB)\n",
            "Collecting termcolor>=1.1.0\n",
            "  Using cached termcolor-1.1.0-py3-none-any.whl\n",
            "Collecting werkzeug>=0.11.15\n",
            "  Using cached Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)\n",
            "Collecting wrapt>=1.11.1\n",
            "  Using cached wrapt-1.12.1-cp36-cp36m-linux_x86_64.whl\n",
            "Collecting alembic\n",
            "  Using cached alembic-1.4.3-py2.py3-none-any.whl (159 kB)\n",
            "Collecting python-editor>=0.3\n",
            "  Using cached python_editor-1.0.4-py3-none-any.whl (4.9 kB)\n",
            "Collecting boto3\n",
            "  Using cached boto3-1.16.43-py2.py3-none-any.whl (130 kB)\n",
            "Collecting botocore<1.20.0,>=1.19.43\n",
            "  Using cached botocore-1.19.43-py2.py3-none-any.whl (7.2 MB)\n",
            "Collecting jmespath<1.0.0,>=0.7.1\n",
            "  Using cached jmespath-0.10.0-py2.py3-none-any.whl (24 kB)\n",
            "Collecting python-dateutil\n",
            "  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)\n",
            "Collecting s3transfer<0.4.0,>=0.3.0\n",
            "  Using cached s3transfer-0.3.3-py2.py3-none-any.whl (69 kB)\n",
            "Collecting cliff\n",
            "  Using cached cliff-3.5.0-py3-none-any.whl (81 kB)\n",
            "Requirement already satisfied: pyparsing>=2.1.0 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from cliff->optuna==1.3->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 5)) (2.4.7)\n",
            "Collecting cmd2!=0.8.3,>=0.8.0\n",
            "  Using cached cmd2-1.4.0-py3-none-any.whl (133 kB)\n",
            "Requirement already satisfied: attrs>=16.3.0 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from cmd2!=0.8.3,>=0.8.0->cliff->optuna==1.3->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 5)) (20.3.0)\n",
            "Requirement already satisfied: wcwidth>=0.1.7 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from cmd2!=0.8.3,>=0.8.0->cliff->optuna==1.3->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 5)) (0.2.5)\n",
            "Collecting colorama>=0.3.7\n",
            "  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/envs/Aspect-Based-Sentiment-Analysis/lib/python3.6/site-packages (from importlib-metadata->markdown>=2.6.8->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->-r /content/Aspect-Based-Sentiment-Analysis/condaenv.zdxk8drz.requirements.txt (line 1)) (3.4.0)\n",
            "Collecting pbr!=2.1.0,>=2.0.0\n",
            "  Using cached pbr-5.5.1-py2.py3-none-any.whl (106 kB)\n",
            "Collecting PrettyTable<0.8,>=0.7.2\n",
            "  Using cached prettytable-0.7.2-py3-none-any.whl\n",
            "Collecting pyperclip>=1.6\n",
            "  Using cached pyperclip-1.8.1-py3-none-any.whl\n",
            "Collecting PyYAML>=3.12\n",
            "  Using cached PyYAML-5.3.1-cp36-cp36m-linux_x86_64.whl\n",
            "Collecting stevedore>=2.0.1\n",
            "  Using cached stevedore-3.3.0-py3-none-any.whl (49 kB)\n",
            "Collecting cmaes\n",
            "  Using cached cmaes-0.7.0-py3-none-any.whl (13 kB)\n",
            "Collecting colorlog\n",
            "  Using cached colorlog-4.6.2-py2.py3-none-any.whl (10.0 kB)\n",
            "Collecting filelock\n",
            "  Using cached filelock-3.0.12-py3-none-any.whl (7.6 kB)\n",
            "Collecting Mako\n",
            "  Using cached Mako-1.1.3-py2.py3-none-any.whl (75 kB)\n",
            "Collecting MarkupSafe>=0.9.2\n",
            "  Using cached MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl (27 kB)\n",
            "Collecting pytz\n",
            "  Using cached pytz-2020.5-py2.py3-none-any.whl (510 kB)\n",
            "Collecting sacremoses\n",
            "  Using cached sacremoses-0.0.43-py3-none-any.whl\n",
            "Collecting click\n",
            "  Using cached click-7.1.2-py2.py3-none-any.whl (82 kB)\n",
            "Collecting sentencepiece\n",
            "  Using cached sentencepiece-0.1.94-cp36-cp36m-manylinux2014_x86_64.whl (1.1 MB)\n",
            "Installing collected packages: pyasn1, rsa, python-dateutil, pyasn1-modules, protobuf, oauthlib, jmespath, cachetools, requests-oauthlib, pytz, pyperclip, pbr, MarkupSafe, googleapis-common-protos, google-auth, colorama, botocore, werkzeug, tensorboard-plugin-wit, stevedore, sqlalchemy, s3transfer, regex, PyYAML, python-editor, PrettyTable, markdown, Mako, grpcio, google-auth-oauthlib, google-api-core, cmd2, click, absl-py, wrapt, tokenizers, termcolor, tensorflow-estimator, tensorboard, sentencepiece, scipy, sacremoses, opt-einsum, keras-preprocessing, h5py, google-resumable-media, google-pasta, google-cloud-core, gast, filelock, colorlog, cmaes, cliff, boto3, astunparse, alembic, transformers, testfixtures, tensorflow, optuna, google-cloud-storage\n",
            "  Attempting uninstall: scipy\n",
            "    Found existing installation: scipy 1.5.2\n",
            "    Uninstalling scipy-1.5.2:\n",
            "      Successfully uninstalled scipy-1.5.2\n",
            "Successfully installed Mako-1.1.3 MarkupSafe-1.1.1 PrettyTable-0.7.2 PyYAML-5.3.1 absl-py-0.11.0 alembic-1.4.3 astunparse-1.6.3 boto3-1.16.43 botocore-1.19.43 cachetools-4.2.0 click-7.1.2 cliff-3.5.0 cmaes-0.7.0 cmd2-1.4.0 colorama-0.4.4 colorlog-4.6.2 filelock-3.0.12 gast-0.3.3 google-api-core-1.24.1 google-auth-1.24.0 google-auth-oauthlib-0.4.2 google-cloud-core-1.5.0 google-cloud-storage-1.26.0 google-pasta-0.2.0 google-resumable-media-0.5.1 googleapis-common-protos-1.52.0 grpcio-1.34.0 h5py-2.10.0 jmespath-0.10.0 keras-preprocessing-1.1.2 markdown-3.3.3 oauthlib-3.1.0 opt-einsum-3.3.0 optuna-1.3.0 pbr-5.5.1 protobuf-3.14.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pyperclip-1.8.1 python-dateutil-2.8.1 python-editor-1.0.4 pytz-2020.5 regex-2020.11.13 requests-oauthlib-1.3.0 rsa-4.6 s3transfer-0.3.3 sacremoses-0.0.43 scipy-1.4.1 sentencepiece-0.1.94 sqlalchemy-1.3.22 stevedore-3.3.0 tensorboard-2.2.2 tensorboard-plugin-wit-1.7.0 tensorflow-2.2.0 tensorflow-estimator-2.2.0 termcolor-1.1.0 testfixtures-6.14.0 tokenizers-0.5.0 transformers-2.5.0 werkzeug-1.0.1 wrapt-1.12.1\n",
            "\n",
            "\b\bdone\n",
            "#\n",
            "# To activate this environment, use\n",
            "#\n",
            "#     $ conda activate Aspect-Based-Sentiment-Analysis\n",
            "#\n",
            "# To deactivate an active environment, use\n",
            "#\n",
            "#     $ conda deactivate\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "R52lQMwqHcz8",
        "outputId": "7c08b38c-ede7-47e0-c04d-0816fb2a0ade"
      },
      "source": [
        "%%bash\r\n",
        "source activate Aspect-Based-Sentiment-Analysis\r\n",
        "python\r\n",
        "import sys\r\n",
        "# maybe only need this the first time we run this notebook\r\n",
        "sys.path.append('/usr/local/lib/python3.6/site-packages')\r\n",
        "\r\n",
        "print(\"Python version\")\r\n",
        "print(sys.version)\r\n"
      ],
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Python version\n",
            "3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31) \n",
            "[GCC 7.3.0]\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dLySYVQAHo1N",
        "outputId": "e57f5aa6-8356-40ae-f412-c6f92c87c7eb"
      },
      "source": [
        "%%writefile /content/Aspect-Based-Sentiment-Analysis/prova.py\r\n",
        "from aspect_based_sentiment_analysis import aux_models\r\n",
        "from aspect_based_sentiment_analysis import utils\r\n",
        "from aspect_based_sentiment_analysis import loads\r\n",
        "from aspect_based_sentiment_analysis import plots\r\n",
        "\r\n",
        "text = (\"the new boss was crazy: he went around the rooms in a crazy way. Employees were aware of that, and they always smile when they used to talk with him. Managers are good people always drink a coffee whit the boss seeming and agree with him, even if they didn't want. This is a perfect day: the boss is ok\")\r\n",
        "recognizer = aux_models.BasicPatternRecognizer()\r\n",
        "nlp = loads.load(pattern_recognizer=recognizer)\r\n",
        "completed_task = nlp(text=text, aspects=['boss', 'manager'])\r\n",
        "subscriptions, price = completed_task.examples\r\n",
        "\r\n",
        "plots.summary(price)\r\n",
        "plots.display(price.review)\r\n",
        "plots.summary(subscriptions)\r\n",
        "plots.display(subscriptions.review)\r\n"
      ],
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Overwriting /content/Aspect-Based-Sentiment-Analysis/prova.py\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "v2jCF_N-HrJQ",
        "outputId": "529e4364-58d7-44e3-cef0-5b3cfce6f1dd"
      },
      "source": [
        "!conda install tensorflow-gpu"
      ],
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bfailed with initial frozen solve. Retrying with flexible solve.\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bfailed with repodata from current_repodata.json, will retry with next repodata source.\n",
            "Collecting package metadata (repodata.json): | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - tensorflow-gpu\n",
            "\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  _tflow_select      pkgs/main/linux-64::_tflow_select-2.1.0-gpu\n",
            "  absl-py            pkgs/main/noarch::absl-py-0.11.0-pyhd3eb1b0_1\n",
            "  astunparse         pkgs/main/noarch::astunparse-1.6.3-py_0\n",
            "  blinker            pkgs/main/linux-64::blinker-1.4-py36_0\n",
            "  c-ares             pkgs/main/linux-64::c-ares-1.17.1-h27cfd23_0\n",
            "  cachetools         pkgs/main/noarch::cachetools-4.2.0-pyhd3eb1b0_0\n",
            "  cudatoolkit        pkgs/main/linux-64::cudatoolkit-10.1.243-h6bb024c_0\n",
            "  cudnn              pkgs/main/linux-64::cudnn-7.6.5-cuda10.1_0\n",
            "  cupti              pkgs/main/linux-64::cupti-10.1.168-0\n",
            "  gast               pkgs/main/noarch::gast-0.3.3-py_0\n",
            "  google-auth        pkgs/main/noarch::google-auth-1.21.3-py_0\n",
            "  google-auth-oauth~ pkgs/main/noarch::google-auth-oauthlib-0.4.2-pyhd3eb1b0_2\n",
            "  google-pasta       pkgs/main/noarch::google-pasta-0.2.0-py_0\n",
            "  grpcio             pkgs/main/linux-64::grpcio-1.31.0-py36hf8bcb03_0\n",
            "  h5py               pkgs/main/linux-64::h5py-2.10.0-py36hd6299e0_1\n",
            "  hdf5               pkgs/main/linux-64::hdf5-1.10.6-hb1b8bf9_0\n",
            "  importlib-metadata pkgs/main/noarch::importlib-metadata-2.0.0-py_1\n",
            "  keras-preprocessi~ pkgs/main/noarch::keras-preprocessing-1.1.0-py_1\n",
            "  libprotobuf        pkgs/main/linux-64::libprotobuf-3.13.0.1-hd408876_0\n",
            "  markdown           pkgs/main/linux-64::markdown-3.3.3-py36h06a4308_0\n",
            "  oauthlib           pkgs/main/noarch::oauthlib-3.1.0-py_0\n",
            "  opt_einsum         pkgs/main/noarch::opt_einsum-3.1.0-py_0\n",
            "  protobuf           pkgs/main/linux-64::protobuf-3.13.0.1-py36he6710b0_1\n",
            "  pyasn1             pkgs/main/noarch::pyasn1-0.4.8-py_0\n",
            "  pyasn1-modules     pkgs/main/noarch::pyasn1-modules-0.2.8-py_0\n",
            "  pyjwt              pkgs/main/linux-64::pyjwt-1.7.1-py36_0\n",
            "  requests-oauthlib  pkgs/main/noarch::requests-oauthlib-1.3.0-py_0\n",
            "  rsa                pkgs/main/noarch::rsa-4.6-py_0\n",
            "  tensorboard        pkgs/main/noarch::tensorboard-2.3.0-pyh4dce500_0\n",
            "  tensorboard-plugi~ pkgs/main/noarch::tensorboard-plugin-wit-1.6.0-py_0\n",
            "  tensorflow         pkgs/main/linux-64::tensorflow-2.2.0-gpu_py36hf933387_0\n",
            "  tensorflow-base    pkgs/main/linux-64::tensorflow-base-2.2.0-gpu_py36h8a81be8_0\n",
            "  tensorflow-estima~ pkgs/main/noarch::tensorflow-estimator-2.2.0-pyh208ff02_0\n",
            "  tensorflow-gpu     pkgs/main/linux-64::tensorflow-gpu-2.2.0-h0d30ee6_0\n",
            "  termcolor          pkgs/main/linux-64::termcolor-1.1.0-py36_1\n",
            "  werkzeug           pkgs/main/noarch::werkzeug-1.0.1-py_0\n",
            "  wrapt              pkgs/main/linux-64::wrapt-1.12.1-py36h7b6447c_1\n",
            "  zipp               pkgs/main/noarch::zipp-3.4.0-pyhd3eb1b0_0\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates    conda-forge::ca-certificates-2020.12.~ --> pkgs/main::ca-certificates-2020.12.8-h06a4308_0\n",
            "\n",
            "The following packages will be SUPERSEDED by a higher-priority channel:\n",
            "\n",
            "  certifi            conda-forge::certifi-2020.12.5-py36h5~ --> pkgs/main::certifi-2020.12.5-py36h06a4308_0\n",
            "  conda              conda-forge::conda-4.9.2-py36h5fab9bb~ --> pkgs/main::conda-4.9.2-py36h06a4308_0\n",
            "\n",
            "\n",
            "Proceed ([y]/n)? y\n",
            "\n",
            "Preparing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ccwL-2O0HtJG",
        "outputId": "c2a470bd-77f6-4adf-d8b0-1c352723ca20"
      },
      "source": [
        "!pip install transformers"
      ],
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: transformers in /usr/local/lib/python3.6/site-packages (2.5.0)\n",
            "Requirement already satisfied: tokenizers==0.5.0 in /usr/local/lib/python3.6/site-packages (from transformers) (0.5.0)\n",
            "Requirement already satisfied: sacremoses in /usr/local/lib/python3.6/site-packages (from transformers) (0.0.43)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.6/site-packages (from transformers) (2020.11.13)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.6/site-packages (from transformers) (3.0.12)\n",
            "Requirement already satisfied: sentencepiece in /usr/local/lib/python3.6/site-packages (from transformers) (0.1.94)\n",
            "Requirement already satisfied: boto3 in /usr/local/lib/python3.6/site-packages (from transformers) (1.16.43)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.6/site-packages (from transformers) (1.19.4)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.6/site-packages (from transformers) (4.54.1)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.6/site-packages (from transformers) (2.25.1)\n",
            "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.6/site-packages (from boto3->transformers) (0.10.0)\n",
            "Requirement already satisfied: s3transfer<0.4.0,>=0.3.0 in /usr/local/lib/python3.6/site-packages (from boto3->transformers) (0.3.3)\n",
            "Requirement already satisfied: botocore<1.20.0,>=1.19.43 in /usr/local/lib/python3.6/site-packages (from boto3->transformers) (1.19.43)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.25.4 in /usr/local/lib/python3.6/site-packages (from botocore<1.20.0,>=1.19.43->boto3->transformers) (1.26.2)\n",
            "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.6/site-packages (from botocore<1.20.0,>=1.19.43->boto3->transformers) (2.8.1)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.6/site-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.20.0,>=1.19.43->boto3->transformers) (1.15.0)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/site-packages (from requests->transformers) (2.10)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.6/site-packages (from requests->transformers) (2020.12.5)\n",
            "Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/lib/python3.6/site-packages (from requests->transformers) (4.0.0)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.6/site-packages (from sacremoses->transformers) (1.0.0)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.6/site-packages (from sacremoses->transformers) (7.1.2)\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dKbTtaNBIMhU",
        "outputId": "9094c69d-c533-4b0c-b43b-59c539680808"
      },
      "source": [
        "!pip install google-cloud-bigquery\r\n"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: google-cloud-bigquery in /usr/local/lib/python3.6/site-packages (2.6.1)\n",
            "Requirement already satisfied: google-api-core[grpc]<2.0.0dev,>=1.23.0 in /usr/local/lib/python3.6/site-packages (from google-cloud-bigquery) (1.24.1)\n",
            "Requirement already satisfied: protobuf>=3.12.0 in /usr/local/lib/python3.6/site-packages (from google-cloud-bigquery) (3.13.0)\n",
            "Requirement already satisfied: proto-plus>=1.10.0 in /usr/local/lib/python3.6/site-packages (from google-cloud-bigquery) (1.13.0)\n",
            "Requirement already satisfied: six<2.0.0dev,>=1.13.0 in /usr/local/lib/python3.6/site-packages (from google-cloud-bigquery) (1.15.0)\n",
            "Requirement already satisfied: google-cloud-core<2.0dev,>=1.4.1 in /usr/local/lib/python3.6/site-packages (from google-cloud-bigquery) (1.5.0)\n",
            "Requirement already satisfied: google-resumable-media<2.0dev,>=0.6.0 in /usr/local/lib/python3.6/site-packages (from google-cloud-bigquery) (1.2.0)\n",
            "Requirement already satisfied: google-auth<2.0dev,>=1.21.1 in /usr/local/lib/python3.6/site-packages (from google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (1.21.3)\n",
            "Requirement already satisfied: requests<3.0.0dev,>=2.18.0 in /usr/local/lib/python3.6/site-packages (from google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (2.25.1)\n",
            "Requirement already satisfied: pytz in /usr/local/lib/python3.6/site-packages (from google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (2020.5)\n",
            "Requirement already satisfied: googleapis-common-protos<2.0dev,>=1.6.0 in /usr/local/lib/python3.6/site-packages (from google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (1.52.0)\n",
            "Requirement already satisfied: setuptools>=34.0.0 in /usr/local/lib/python3.6/site-packages (from google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (51.0.0.post20201207)\n",
            "Requirement already satisfied: grpcio<2.0dev,>=1.29.0 in /usr/local/lib/python3.6/site-packages (from google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (1.31.0)\n",
            "Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.6/site-packages (from google-auth<2.0dev,>=1.21.1->google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (4.6)\n",
            "Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.6/site-packages (from google-auth<2.0dev,>=1.21.1->google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (0.2.8)\n",
            "Requirement already satisfied: cachetools<5.0,>=2.0.0 in /usr/local/lib/python3.6/site-packages (from google-auth<2.0dev,>=1.21.1->google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (4.2.0)\n",
            "Requirement already satisfied: google-crc32c<2.0dev,>=1.0 in /usr/local/lib/python3.6/site-packages (from google-resumable-media<2.0dev,>=0.6.0->google-cloud-bigquery) (1.1.0)\n",
            "Requirement already satisfied: cffi>=1.0.0 in /usr/local/lib/python3.6/site-packages (from google-crc32c<2.0dev,>=1.0->google-resumable-media<2.0dev,>=0.6.0->google-cloud-bigquery) (1.14.4)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.6/site-packages (from cffi>=1.0.0->google-crc32c<2.0dev,>=1.0->google-resumable-media<2.0dev,>=0.6.0->google-cloud-bigquery) (2.20)\n",
            "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /usr/local/lib/python3.6/site-packages (from pyasn1-modules>=0.2.1->google-auth<2.0dev,>=1.21.1->google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (0.4.8)\n",
            "Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/lib/python3.6/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (4.0.0)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.6/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (1.26.2)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.6/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (2020.12.5)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/site-packages (from requests<3.0.0dev,>=2.18.0->google-api-core[grpc]<2.0.0dev,>=1.23.0->google-cloud-bigquery) (2.10)\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eGKLsXsLISnV",
        "outputId": "02e0b91a-f4c9-4a72-9e4e-d87f91c98434"
      },
      "source": [
        "!pip install aspect-based-sentiment-analysis"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: aspect-based-sentiment-analysis in /usr/local/lib/python3.6/site-packages (2.0.2)\n",
            "Requirement already satisfied: optuna in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (2.3.0)\n",
            "Requirement already satisfied: ipython in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (7.16.1)\n",
            "Requirement already satisfied: testfixtures in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (6.17.0)\n",
            "Requirement already satisfied: transformers==2.5 in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (2.5.0)\n",
            "Requirement already satisfied: pytest in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (6.2.1)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (0.24.0)\n",
            "Requirement already satisfied: tensorflow==2.2 in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (2.2.0)\n",
            "Requirement already satisfied: google-cloud-storage in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (1.35.0)\n",
            "Requirement already satisfied: spacy in /usr/local/lib/python3.6/site-packages (from aspect-based-sentiment-analysis) (2.3.5)\n",
            "Requirement already satisfied: protobuf>=3.8.0 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (3.13.0)\n",
            "Requirement already satisfied: gast==0.3.3 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (0.3.3)\n",
            "Requirement already satisfied: keras-preprocessing>=1.1.0 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (1.1.0)\n",
            "Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (3.1.0)\n",
            "Requirement already satisfied: wrapt>=1.11.1 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (1.12.1)\n",
            "Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (1.15.0)\n",
            "Requirement already satisfied: google-pasta>=0.1.8 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (0.2.0)\n",
            "Requirement already satisfied: tensorflow-estimator<2.3.0,>=2.2.0 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (2.2.0)\n",
            "Requirement already satisfied: h5py<2.11.0,>=2.10.0 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (2.10.0)\n",
            "Requirement already satisfied: absl-py>=0.7.0 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (0.11.0)\n",
            "Requirement already satisfied: grpcio>=1.8.6 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (1.31.0)\n",
            "Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (1.1.0)\n",
            "Requirement already satisfied: wheel>=0.26 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (0.36.2)\n",
            "Requirement already satisfied: astunparse==1.6.3 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (1.6.3)\n",
            "Requirement already satisfied: numpy<2.0,>=1.16.0 in /usr/local/lib/python3.6/site-packages (from tensorflow==2.2->aspect-based-sentiment-analysis) (1.19.4)\n",
            "Collecting scipy==1.4.1\n",
            "  Using cached scipy-1.4.1-cp36-cp36m-manylinux1_x86_64.whl (26.1 MB)\n",
            "Requirement already satisfied: tokenizers==0.5.0 in /usr/local/lib/python3.6/site-packages (from transformers==2.5->aspect-based-sentiment-analysis) (0.5.0)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.6/site-packages (from transformers==2.5->aspect-based-sentiment-analysis) (4.54.1)\n",
            "Requirement already satisfied: sentencepiece in /usr/local/lib/python3.6/site-packages (from transformers==2.5->aspect-based-sentiment-analysis) (0.1.94)\n",
            "Requirement already satisfied: sacremoses in /usr/local/lib/python3.6/site-packages (from transformers==2.5->aspect-based-sentiment-analysis) (0.0.43)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.6/site-packages (from transformers==2.5->aspect-based-sentiment-analysis) (2.25.1)\n",
            "Requirement already satisfied: boto3 in /usr/local/lib/python3.6/site-packages (from transformers==2.5->aspect-based-sentiment-analysis) (1.16.43)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.6/site-packages (from transformers==2.5->aspect-based-sentiment-analysis) (3.0.12)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.6/site-packages (from transformers==2.5->aspect-based-sentiment-analysis) (2020.11.13)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from protobuf>=3.8.0->tensorflow==2.2->aspect-based-sentiment-analysis) (51.0.0.post20201207)\n",
            "Collecting tensorboard<2.3.0,>=2.2.0\n",
            "  Using cached tensorboard-2.2.2-py3-none-any.whl (3.0 MB)\n",
            "Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /usr/local/lib/python3.6/site-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (1.6.0)\n",
            "Requirement already satisfied: werkzeug>=0.11.15 in /usr/local/lib/python3.6/site-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (1.0.1)\n",
            "Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /usr/local/lib/python3.6/site-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (0.4.2)\n",
            "Requirement already satisfied: google-auth<2,>=1.6.3 in /usr/local/lib/python3.6/site-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (1.21.3)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.6/site-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (3.3.3)\n",
            "Requirement already satisfied: cachetools<5.0,>=2.0.0 in /usr/local/lib/python3.6/site-packages (from google-auth<2,>=1.6.3->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (4.2.0)\n",
            "Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.6/site-packages (from google-auth<2,>=1.6.3->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (0.2.8)\n",
            "Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.6/site-packages (from google-auth<2,>=1.6.3->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (4.6)\n",
            "Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/local/lib/python3.6/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (1.3.0)\n",
            "Requirement already satisfied: importlib-metadata in /usr/local/lib/python3.6/site-packages (from markdown>=2.6.8->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (2.0.0)\n",
            "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /usr/local/lib/python3.6/site-packages (from pyasn1-modules>=0.2.1->google-auth<2,>=1.6.3->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (0.4.8)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/site-packages (from requests->transformers==2.5->aspect-based-sentiment-analysis) (2.10)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.6/site-packages (from requests->transformers==2.5->aspect-based-sentiment-analysis) (1.26.2)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.6/site-packages (from requests->transformers==2.5->aspect-based-sentiment-analysis) (2020.12.5)\n",
            "Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/lib/python3.6/site-packages (from requests->transformers==2.5->aspect-based-sentiment-analysis) (4.0.0)\n",
            "Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.6/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (3.1.0)\n",
            "Requirement already satisfied: s3transfer<0.4.0,>=0.3.0 in /usr/local/lib/python3.6/site-packages (from boto3->transformers==2.5->aspect-based-sentiment-analysis) (0.3.3)\n",
            "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.6/site-packages (from boto3->transformers==2.5->aspect-based-sentiment-analysis) (0.10.0)\n",
            "Requirement already satisfied: botocore<1.20.0,>=1.19.43 in /usr/local/lib/python3.6/site-packages (from boto3->transformers==2.5->aspect-based-sentiment-analysis) (1.19.43)\n",
            "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.6/site-packages (from botocore<1.20.0,>=1.19.43->boto3->transformers==2.5->aspect-based-sentiment-analysis) (2.8.1)\n",
            "Requirement already satisfied: google-cloud-core<2.0dev,>=1.4.1 in /usr/local/lib/python3.6/site-packages (from google-cloud-storage->aspect-based-sentiment-analysis) (1.5.0)\n",
            "Requirement already satisfied: google-resumable-media<2.0dev,>=1.2.0 in /usr/local/lib/python3.6/site-packages (from google-cloud-storage->aspect-based-sentiment-analysis) (1.2.0)\n",
            "Requirement already satisfied: google-api-core<2.0.0dev,>=1.21.0 in /usr/local/lib/python3.6/site-packages (from google-cloud-core<2.0dev,>=1.4.1->google-cloud-storage->aspect-based-sentiment-analysis) (1.24.1)\n",
            "Requirement already satisfied: pytz in /usr/local/lib/python3.6/site-packages (from google-api-core<2.0.0dev,>=1.21.0->google-cloud-core<2.0dev,>=1.4.1->google-cloud-storage->aspect-based-sentiment-analysis) (2020.5)\n",
            "Requirement already satisfied: googleapis-common-protos<2.0dev,>=1.6.0 in /usr/local/lib/python3.6/site-packages (from google-api-core<2.0.0dev,>=1.21.0->google-cloud-core<2.0dev,>=1.4.1->google-cloud-storage->aspect-based-sentiment-analysis) (1.52.0)\n",
            "Requirement already satisfied: google-crc32c<2.0dev,>=1.0 in /usr/local/lib/python3.6/site-packages (from google-resumable-media<2.0dev,>=1.2.0->google-cloud-storage->aspect-based-sentiment-analysis) (1.1.0)\n",
            "Requirement already satisfied: cffi>=1.0.0 in /usr/local/lib/python3.6/site-packages (from google-crc32c<2.0dev,>=1.0->google-resumable-media<2.0dev,>=1.2.0->google-cloud-storage->aspect-based-sentiment-analysis) (1.14.4)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.6/site-packages (from cffi>=1.0.0->google-crc32c<2.0dev,>=1.0->google-resumable-media<2.0dev,>=1.2.0->google-cloud-storage->aspect-based-sentiment-analysis) (2.20)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.6/site-packages (from importlib-metadata->markdown>=2.6.8->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2->aspect-based-sentiment-analysis) (3.4.0)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.6/site-packages (from ipython->aspect-based-sentiment-analysis) (2.7.3)\n",
            "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.6/site-packages (from ipython->aspect-based-sentiment-analysis) (3.0.8)\n",
            "Requirement already satisfied: jedi>=0.10 in /usr/local/lib/python3.6/site-packages (from ipython->aspect-based-sentiment-analysis) (0.18.0)\n",
            "Requirement already satisfied: pexpect in /usr/local/lib/python3.6/site-packages (from ipython->aspect-based-sentiment-analysis) (4.8.0)\n",
            "Requirement already satisfied: backcall in /usr/local/lib/python3.6/site-packages (from ipython->aspect-based-sentiment-analysis) (0.2.0)\n",
            "Requirement already satisfied: traitlets>=4.2 in /usr/local/lib/python3.6/site-packages (from ipython->aspect-based-sentiment-analysis) (4.3.3)\n",
            "Requirement already satisfied: decorator in /usr/local/lib/python3.6/site-packages (from ipython->aspect-based-sentiment-analysis) (4.4.2)\n",
            "Requirement already satisfied: pickleshare in /usr/local/lib/python3.6/site-packages (from ipython->aspect-based-sentiment-analysis) (0.7.5)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.6/site-packages (from jedi>=0.10->ipython->aspect-based-sentiment-analysis) (0.8.1)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.6/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython->aspect-based-sentiment-analysis) (0.2.5)\n",
            "Requirement already satisfied: ipython-genutils in /usr/local/lib/python3.6/site-packages (from traitlets>=4.2->ipython->aspect-based-sentiment-analysis) (0.2.0)\n",
            "Requirement already satisfied: colorlog in /usr/local/lib/python3.6/site-packages (from optuna->aspect-based-sentiment-analysis) (4.6.2)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.6/site-packages (from optuna->aspect-based-sentiment-analysis) (1.0.0)\n",
            "Requirement already satisfied: sqlalchemy>=1.1.0 in /usr/local/lib/python3.6/site-packages (from optuna->aspect-based-sentiment-analysis) (1.3.22)\n",
            "Requirement already satisfied: cliff in /usr/local/lib/python3.6/site-packages (from optuna->aspect-based-sentiment-analysis) (3.5.0)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.6/site-packages (from optuna->aspect-based-sentiment-analysis) (20.8)\n",
            "Requirement already satisfied: cmaes>=0.6.0 in /usr/local/lib/python3.6/site-packages (from optuna->aspect-based-sentiment-analysis) (0.7.0)\n",
            "Requirement already satisfied: alembic in /usr/local/lib/python3.6/site-packages (from optuna->aspect-based-sentiment-analysis) (1.4.3)\n",
            "Requirement already satisfied: pyparsing>=2.0.2 in /usr/local/lib/python3.6/site-packages (from packaging>=20.0->optuna->aspect-based-sentiment-analysis) (2.4.7)\n",
            "Requirement already satisfied: python-editor>=0.3 in /usr/local/lib/python3.6/site-packages (from alembic->optuna->aspect-based-sentiment-analysis) (1.0.4)\n",
            "Requirement already satisfied: Mako in /usr/local/lib/python3.6/site-packages (from alembic->optuna->aspect-based-sentiment-analysis) (1.1.3)\n",
            "Requirement already satisfied: cmd2!=0.8.3,>=0.8.0 in /usr/local/lib/python3.6/site-packages (from cliff->optuna->aspect-based-sentiment-analysis) (1.4.0)\n",
            "Requirement already satisfied: pbr!=2.1.0,>=2.0.0 in /usr/local/lib/python3.6/site-packages (from cliff->optuna->aspect-based-sentiment-analysis) (5.5.1)\n",
            "Requirement already satisfied: PrettyTable<0.8,>=0.7.2 in /usr/local/lib/python3.6/site-packages (from cliff->optuna->aspect-based-sentiment-analysis) (0.7.2)\n",
            "Requirement already satisfied: stevedore>=2.0.1 in /usr/local/lib/python3.6/site-packages (from cliff->optuna->aspect-based-sentiment-analysis) (3.3.0)\n",
            "Requirement already satisfied: PyYAML>=3.12 in /usr/local/lib/python3.6/site-packages (from cliff->optuna->aspect-based-sentiment-analysis) (5.3.1)\n",
            "Requirement already satisfied: colorama>=0.3.7 in /usr/local/lib/python3.6/site-packages (from cmd2!=0.8.3,>=0.8.0->cliff->optuna->aspect-based-sentiment-analysis) (0.4.4)\n",
            "Requirement already satisfied: attrs>=16.3.0 in /usr/local/lib/python3.6/site-packages (from cmd2!=0.8.3,>=0.8.0->cliff->optuna->aspect-based-sentiment-analysis) (20.3.0)\n",
            "Requirement already satisfied: pyperclip>=1.6 in /usr/local/lib/python3.6/site-packages (from cmd2!=0.8.3,>=0.8.0->cliff->optuna->aspect-based-sentiment-analysis) (1.8.1)\n",
            "Requirement already satisfied: MarkupSafe>=0.9.2 in /usr/local/lib/python3.6/site-packages (from Mako->alembic->optuna->aspect-based-sentiment-analysis) (1.1.1)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.6/site-packages (from pexpect->ipython->aspect-based-sentiment-analysis) (0.6.0)\n",
            "Requirement already satisfied: iniconfig in /usr/local/lib/python3.6/site-packages (from pytest->aspect-based-sentiment-analysis) (1.1.1)\n",
            "Requirement already satisfied: toml in /usr/local/lib/python3.6/site-packages (from pytest->aspect-based-sentiment-analysis) (0.10.2)\n",
            "Requirement already satisfied: py>=1.8.2 in /usr/local/lib/python3.6/site-packages (from pytest->aspect-based-sentiment-analysis) (1.10.0)\n",
            "Requirement already satisfied: pluggy<1.0.0a1,>=0.12 in /usr/local/lib/python3.6/site-packages (from pytest->aspect-based-sentiment-analysis) (0.13.1)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.6/site-packages (from sacremoses->transformers==2.5->aspect-based-sentiment-analysis) (7.1.2)\n",
            "Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.6/site-packages (from scikit-learn->aspect-based-sentiment-analysis) (2.1.0)\n",
            "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (1.0.5)\n",
            "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (3.0.5)\n",
            "Requirement already satisfied: blis<0.8.0,>=0.4.0 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (0.7.4)\n",
            "Requirement already satisfied: catalogue<1.1.0,>=0.0.7 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (1.0.0)\n",
            "Requirement already satisfied: plac<1.2.0,>=0.9.6 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (1.1.3)\n",
            "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (2.0.5)\n",
            "Requirement already satisfied: wasabi<1.1.0,>=0.4.0 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (0.8.0)\n",
            "Requirement already satisfied: srsly<1.1.0,>=1.0.2 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (1.0.5)\n",
            "Requirement already satisfied: thinc<7.5.0,>=7.4.1 in /usr/local/lib/python3.6/site-packages (from spacy->aspect-based-sentiment-analysis) (7.4.5)\n",
            "Installing collected packages: tensorboard, scipy\n",
            "  Attempting uninstall: tensorboard\n",
            "    Found existing installation: tensorboard 2.3.0\n",
            "    Uninstalling tensorboard-2.3.0:\n",
            "      Successfully uninstalled tensorboard-2.3.0\n",
            "  Attempting uninstall: scipy\n",
            "    Found existing installation: scipy 1.5.3\n",
            "    Uninstalling scipy-1.5.3:\n",
            "      Successfully uninstalled scipy-1.5.3\n",
            "Successfully installed scipy-1.4.1 tensorboard-2.2.2\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ctHYVrHRIRGp",
        "outputId": "c6249d02-dc7c-45f0-b104-2203dd9ea021"
      },
      "source": [
        "!python /content/Aspect-Based-Sentiment-Analysis/prova.py"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "2020-12-27 18:47:53.656748: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1\n",
            "2020-12-27 18:47:53.690844: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.691465: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n",
            "pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n",
            "coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n",
            "2020-12-27 18:47:53.693576: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 18:47:53.704815: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "2020-12-27 18:47:53.712552: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n",
            "2020-12-27 18:47:53.719599: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n",
            "2020-12-27 18:47:53.728360: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n",
            "2020-12-27 18:47:53.733137: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n",
            "2020-12-27 18:47:53.738893: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n",
            "2020-12-27 18:47:53.739073: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.739716: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.740271: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n",
            "2020-12-27 18:47:53.740665: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
            "2020-12-27 18:47:53.751499: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2200000000 Hz\n",
            "2020-12-27 18:47:53.751771: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x559c7a6c2540 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n",
            "2020-12-27 18:47:53.751824: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n",
            "2020-12-27 18:47:53.752043: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.752642: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n",
            "pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n",
            "coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n",
            "2020-12-27 18:47:53.752717: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 18:47:53.752757: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "2020-12-27 18:47:53.752787: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n",
            "2020-12-27 18:47:53.752822: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n",
            "2020-12-27 18:47:53.752850: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n",
            "2020-12-27 18:47:53.752878: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n",
            "2020-12-27 18:47:53.752906: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n",
            "2020-12-27 18:47:53.752993: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.753622: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.754228: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n",
            "2020-12-27 18:47:53.754331: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 18:47:53.868998: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
            "2020-12-27 18:47:53.869049: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 \n",
            "2020-12-27 18:47:53.869067: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N \n",
            "2020-12-27 18:47:53.869259: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.869976: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.870631: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:47:53.871247: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n",
            "2020-12-27 18:47:53.871310: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 14071 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:00:04.0, compute capability: 7.5)\n",
            "2020-12-27 18:47:53.873108: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x559c7a6c3500 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:\n",
            "2020-12-27 18:47:53.873147: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla T4, Compute Capability 7.5\n",
            "2020-12-27 18:47:54.558835: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "Sentiment.positive for \"manager\"\n",
            "Scores (neutral/negative/positive): [0.01  0.029 0.96 ]\n",
            "<IPython.core.display.HTML object>\n",
            "Sentiment.negative for \"boss\"\n",
            "Scores (neutral/negative/positive): [0.137 0.744 0.119]\n",
            "<IPython.core.display.HTML object>\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tIU3LRCGONSi",
        "outputId": "2d046792-e93d-4eca-81b0-b19d6c2682cc"
      },
      "source": [
        "!python /content/Aspect-Based-Sentiment-Analysis/examples/train_classifier.py --domain restaurant --n_trials 1"
      ],
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32m[I 2020-12-27 18:51:02,547]\u001b[0m Using an existing study with name 'classifier-restaurant' instead of creating a new one.\u001b[0m\n",
            "2020-12-27 18:51:03.546171: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1\n",
            "2020-12-27 18:51:03.581310: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.581901: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n",
            "pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n",
            "coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n",
            "2020-12-27 18:51:03.582211: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 18:51:03.584182: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "2020-12-27 18:51:03.585790: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n",
            "2020-12-27 18:51:03.586107: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n",
            "2020-12-27 18:51:03.588092: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n",
            "2020-12-27 18:51:03.590043: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n",
            "2020-12-27 18:51:03.595194: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n",
            "2020-12-27 18:51:03.595360: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.596242: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.596993: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n",
            "2020-12-27 18:51:03.597380: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
            "2020-12-27 18:51:03.606223: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2200000000 Hz\n",
            "2020-12-27 18:51:03.606431: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55a15e091800 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n",
            "2020-12-27 18:51:03.606461: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n",
            "2020-12-27 18:51:03.606646: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.607462: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n",
            "pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n",
            "coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n",
            "2020-12-27 18:51:03.607538: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 18:51:03.607570: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "2020-12-27 18:51:03.607596: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n",
            "2020-12-27 18:51:03.607620: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n",
            "2020-12-27 18:51:03.607642: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n",
            "2020-12-27 18:51:03.607666: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n",
            "2020-12-27 18:51:03.607729: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n",
            "2020-12-27 18:51:03.607837: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.608852: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.609933: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n",
            "2020-12-27 18:51:03.610116: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 18:51:03.725869: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
            "2020-12-27 18:51:03.725911: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 \n",
            "2020-12-27 18:51:03.725933: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N \n",
            "2020-12-27 18:51:03.726110: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.726746: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.727333: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 18:51:03.727856: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n",
            "2020-12-27 18:51:03.727902: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 14071 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:00:04.0, compute capability: 7.5)\n",
            "2020-12-27 18:51:03.729393: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55a15e090bc0 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:\n",
            "2020-12-27 18:51:03.729422: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla T4, Compute Capability 7.5\n",
            "2020-12-27 18:51:04.431118: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "2020-12-27 18:51:06 [INFO  ] [absa.callbacks] Make a checkpoint directory: /content/Aspect-Based-Sentiment-Analysis/examples/optimization/classifier-restaurant-115/checkpoints\n",
            "2020-12-27 18:51:56 [INFO  ] [absa.callbacks] Epoch    1 Accuracy   Average Train 0.738    Average Test 0.804\n",
            "2020-12-27 18:51:58 [INFO  ] [absa.callbacks] New model checkpoint saved.\n",
            "2020-12-27 18:52:49 [INFO  ] [absa.callbacks] Epoch    2 Accuracy   Average Train 0.855    Average Test 0.838\n",
            "2020-12-27 18:52:50 [INFO  ] [absa.callbacks] New model checkpoint saved.\n",
            "2020-12-27 18:53:42 [INFO  ] [absa.callbacks] Epoch    3 Accuracy   Average Train 0.915    Average Test 0.852\n",
            "2020-12-27 18:53:43 [INFO  ] [absa.callbacks] New model checkpoint saved.\n",
            "2020-12-27 18:54:34 [INFO  ] [absa.callbacks] Epoch    4 Accuracy   Average Train 0.950    Average Test 0.835\n",
            "2020-12-27 18:55:26 [INFO  ] [absa.callbacks] Epoch    5 Accuracy   Average Train 0.968    Average Test 0.855\n",
            "2020-12-27 18:55:27 [INFO  ] [absa.callbacks] New model checkpoint saved.\n",
            "2020-12-27 18:56:19 [INFO  ] [absa.callbacks] Epoch    6 Accuracy   Average Train 0.970    Average Test 0.852\n",
            "2020-12-27 18:56:19 [INFO  ] [absa.routines] The training routine is stopped.\n",
            "\u001b[32m[I 2020-12-27 18:56:22,740]\u001b[0m Trial 115 finished with value: 0.8551136255264282 and parameters: {'batch_size': 32, 'learning_rate': 7.565657372890847e-05, 'beta_1': 0.5999765715849746}. Best is trial 67 with value: 0.9083333611488342.\u001b[0m\n",
            "\u001b[0m"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Y8_ZryJxT7Iq",
        "outputId": "c997cb53-777f-41a9-cad2-1a34091d5bd2"
      },
      "source": [
        "#necessari per poter avviare la fase di test, i seguenti comandi!\r\n",
        "#!python -m spacy download en\r\n",
        "#import spacy\r\n",
        "#spacy.load('en')"
      ],
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<spacy.lang.en.English at 0x7efcbdb5eb38>"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NMQ4Xbs2QgrI",
        "outputId": "c3adac13-42ca-4335-bede-c3c4b5b890e9"
      },
      "source": [
        "!python /content/Aspect-Based-Sentiment-Analysis/tests/test_performance.py"
      ],
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "2020-12-27 19:07:23.442487: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1\n",
            "2020-12-27 19:07:23.484076: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.484632: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n",
            "pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n",
            "coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n",
            "2020-12-27 19:07:23.484901: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 19:07:23.486872: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "2020-12-27 19:07:23.488516: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n",
            "2020-12-27 19:07:23.488841: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n",
            "2020-12-27 19:07:23.490807: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n",
            "2020-12-27 19:07:23.491999: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n",
            "2020-12-27 19:07:23.496139: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n",
            "2020-12-27 19:07:23.496256: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.496844: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.497345: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n",
            "2020-12-27 19:07:23.497639: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
            "2020-12-27 19:07:23.503049: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2200000000 Hz\n",
            "2020-12-27 19:07:23.503254: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x56376ed0d500 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n",
            "2020-12-27 19:07:23.503285: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n",
            "2020-12-27 19:07:23.503465: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.504059: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n",
            "pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n",
            "coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n",
            "2020-12-27 19:07:23.504110: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 19:07:23.504145: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "2020-12-27 19:07:23.504170: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n",
            "2020-12-27 19:07:23.504193: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n",
            "2020-12-27 19:07:23.504215: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n",
            "2020-12-27 19:07:23.504236: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n",
            "2020-12-27 19:07:23.504258: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n",
            "2020-12-27 19:07:23.504329: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.504928: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.505422: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n",
            "2020-12-27 19:07:23.505482: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n",
            "2020-12-27 19:07:23.611729: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
            "2020-12-27 19:07:23.611771: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 \n",
            "2020-12-27 19:07:23.611785: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N \n",
            "2020-12-27 19:07:23.611956: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.612553: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.613113: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2020-12-27 19:07:23.613604: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n",
            "2020-12-27 19:07:23.613652: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 14071 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:00:04.0, compute capability: 7.5)\n",
            "2020-12-27 19:07:23.615113: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x56376ed0ca80 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:\n",
            "2020-12-27 19:07:23.615144: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla T4, Compute Capability 7.5\n",
            "2020-12-27 19:07:24.295066: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n",
            "--------------------------------------------------------------------------\n",
            "--------------------------------------------------------------------------\n",
            "0.8607142857142858\n",
            "--------------------------------------------------------------------------\n",
            "--------------------------------------------------------------------------\n",
            "Pipeline(model=<aspect_based_sentiment_analysis.models.BertABSClassifier object at 0x7f16bf24ee48>, tokenizer=<transformers.tokenization_bert.BertTokenizer object at 0x7f16be8574a8>, professor=Professor(reference_recognizer=None, pattern_recognizer=None), text_splitter=<function sentencizer.<locals>.wrapper at 0x7f172550b0d0>)\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "iTO0Rox5nUeP"
      },
      "source": [
        "#!python /content/Aspect-Based-Sentiment-Analysis/tests/data/test_semeval.py"
      ],
      "execution_count": 19,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "NMOQA48dnh2N"
      },
      "source": [
        "!cp /content/Aspect-Based-Sentiment-Analysis/data/semeval/adapter.py /content/Aspect-Based-Sentiment-Analysis/tests/data"
      ],
      "execution_count": 13,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "d71Xt5CnfLqw"
      },
      "source": [
        ""
      ]
    }
  ]
}